File: backend/StudioManagement.Data/Entities/Customer.cs
namespace StudioManagement.Data.Entities;

public class Customer : ITenantEntity
{
    public int CustomerId { get; set; }
    public int StudioId { get; set; }
    public string FullName { get; set; } = null!;
    public string MobileNumber { get; set; } = null!;
    public string? Email { get; set; }
    public string? Address { get; set; }
    public string? Notes { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Studio Studio { get; set; } = null!;
    public ICollection<Event> Events { get; set; } = new List<Event>();
    public ICollection<Quotation> Quotations { get; set; } = new List<Quotation>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
}

File: frontend/src/types/customer.ts
export interface Customer {
  customerId: number;
  fullName: string;
  mobileNumber: string;
  email: string | null;
  address: string | null;
  notes: string | null;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateCustomerRequest {
  fullName: string;
  mobileNumber: string;
  email?: string;
  address?: string;
  notes?: string;
}

export type UpdateCustomerRequest = CreateCustomerRequest;

export interface CustomerEventSummary {
  eventId: number;
  eventTypeName: string | null;
  eventDate: string;
  startTime: string | null;
  endTime: string | null;
  venue: string | null;
  venueAddress: string | null;
  eventStatus: string;
  budget: number | null;
  amountPaid: number;
  balance: number;
  workerCount: number;
  notes: string | null;
}
